src/go/rdctl/pkg/snapshot/manager_test.go

```diff
@@ -42,55 +42,105 @@ func TestManager(t *testing.T) {
+			ExpectedError: `invalid character '\r' at position 16 in name: all characters must be printable or a space`,
+		},
+		{
+			Name:          "can't contain a \x00 null-byte",
+			ExpectedError: `invalid character '\x00' at position 16 in name: all characters must be printable or a space`,
+		},
+		{
+			Name:          "can't contain a \a control character",
+			ExpectedError: `invalid character '\a' at position 16 in name: all characters must be printable or a space`,
+		},
+	}
+	for _, testCase := range testCases {
+		description := fmt.Sprintf("ValidateName should disallow invalid names (case %+v)", testCase)
+		t.Run(description, func(t *testing.T) {
+			paths, _ := populateFiles(t, true)
+			manager := newTestManager(paths)
+			// The test case name is used in the name of a file inside the temporary
+			// directory, which means it is limited in length. Work around this.
+			if testCase.Name == "filename_too_long_workaround" {
+				// 251 characters is too long (and the indentation here is what our linter demands)
+				testCase.Name = "12345678911234567892123456789312345678941234567895123456789612345678971234567898" +
+					"12345678991234567890123456789112345678921234567893123456789412345678951234567896" +
+					"12345678971234567898123456789912345678901234567891123456789212345678931234567894" +
+					"12345678951"
 			}
-		}
-		// The above loop verifies that each invalid name is found invalid.
-		// The following code verifies that we get the actual
-		err := manager.ValidateName(oversizeName)
-		if err == nil {
-			t.Error("oversize name is invalid but no error was returned")
-		} else if !strings.Contains(err.Error(), "…") {
-			t.Errorf("No ellipsis in reported name")
-		}
-		err = manager.ValidateName(nameWithTab)
-		if err == nil {
-			t.Error("name with tab is invalid but no error was returned")
-		} else if !strings.Contains(err.Error(), "invalid character value 9 at position 16 in name: all characters must be printable or a space") {
-			t.Errorf("failed to report unprintable character in name, got %s", err.Error())
-		}
-		longishNameEndingWithSpace := "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ "
-		endOfLongishString := longishNameEndingWithSpace[len(longishNameEndingWithSpace)-nameDisplayCutoffSize:]
-		err = manager.ValidateName(longishNameEndingWithSpace)
-		if err == nil {
-			t.Error(" name ending with space not caught")
-		} else if !strings.Contains(err.Error(), "…"+endOfLongishString) {
-			t.Errorf("Longish invalid name not truncated as expected")
-		}
-	})
+			err := manager.ValidateName(testCase.Name)
+			if err == nil {
+				t.Errorf("expected error but got err == nil")
+			} else if !strings.Contains(err.Error(), testCase.ExpectedError) {
+				t.Errorf("unexpected error %q", err)
+			}
+			// check that we are truncating the name properly in the error message
+			if len(testCase.ExpectedErrMsgName) > 0 {
+				if !strings.Contains(err.Error(), testCase.ExpectedErrMsgName) {
```

Comment:
I experimented with this a little bit, but my head started to spin with all the quoting inside quoting, and the associated backslash escapes. I'd prefer to leave it as is.

Reply:
One reason I was suggesting it is to make sure this test case includes the leading space in the error message:

```golang
		{
			Name:               " 12345678911234567892123456我喜欢鸡肉",
			ExpectedError:      "must not start with a white-space character",
			ExpectedErrMsgName: "12345678911234567892123456我喜…",
		},
```

The current code would match:

```
invalid name "12345678911234567892123456我喜…": must not start with a white-space character
```

which would be confusing (where is the white-space character). This could happen if we called the other (now renamed) `truncate` function. So we really expect the whitespace to be visible:

```
invalid name " 12345678911234567892123456我喜…": must not start with a white-space character
```

and calling `strconv.Quote()` should do that to match the `%q` formatting.